typescript/middy.d.ts

```diff
@@ -0,0 +1,35 @@
+import {Callback, Context, Handler, ProxyResult} from 'aws-lambda';
+
+export interface IMiddy {
+  use: IMiddyUseFunction;
+  before: IMiddyMiddlewareFunction;
+  after: IMiddyMiddlewareFunction;
+  onError: IMiddyMiddlewareFunction;
+}
+
+export type IMiddyUseFunction = (config?: object) => IMiddy;
+
+export interface IMiddyMiddlewareObject {
+  before?: IMiddyMiddlewareFunction;
+  after?: IMiddyMiddlewareFunction;
+  onError?: IMiddyMiddlewareFunction;
+}
+
+type IMiddyMiddlewareFunction = (
+  handler: IHandlerLambda,
+  next: IMiddyNextFunction
+) => void;
```

Comment:
This, from version 0.3.0, also supports promises as return value (as an alternative to calling the callback directly). Is there anyway we can make this explicit?